src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -646,7 +646,12 @@ public function dumpFile($filename, $content)
             throw new IOException(sprintf('Failed to write file "%s".', $filename), 0, null, $filename);
         }
 
-        @chmod($tmpFile, 0666 & ~umask());
```

Comment:
It may depend on what is considered the most readable.
By the way, what about `$this -> chmod()` instead of `chmodd()` ?